base/applications/regedit/find.c

```diff
@@ -223,14 +223,28 @@ BOOL RegFindRecurse(
                                   NULL, &cb);
         if (lResult != ERROR_SUCCESS)
             goto err;
-        pb = malloc(cb);
+        pb = malloc(cb + 3); /* To avoid buffer overrun, append 3 NULs */
         if (pb == NULL)
             goto err;
         lResult = RegQueryValueExW(hSubKey, ppszNames[i], NULL, &type,
                                   pb, &cb);
         if (lResult != ERROR_SUCCESS)
             goto err;
 
+        if (type == REG_SZ || type == REG_EXPAND_SZ)
+        {
+            /* Test for odd (not even) number of bytes returned */
+            if (cb & 1)
+                DPRINTF("Byte Count of (%d) for wide string is ODD (Not even)\n", cb);
+        }
+
+        /* To avoid buffer overrun, append 3 NUL bytes.
+           NOTE: cb can be an odd number although UNICODE_NULL is two bytes.
+           Two bytes at odd position is not enough to avoid buffer overrun. */
+        pb[cb] = pb[cb + 1] = pb[cb + 2] = 0;
+
+        /* Now pb is guaranteed not to cause buffer overrun. */
```

Comment:
```suggestion
```
I think this is clear from the previous comment ^^

Reply:
Removed. Thanks.